### ISSUE TITLE
Copy/extract Highlighted Fields

### ISSUE THREAD

**Issue by felipecorral** [RobHolme/vscode-hl7tools#6]
Hi Rob,
I am trying to extract a field of a multiple HL7 messages,
actually i can Higlight the fields using "HL7 Tools: Highlight Field", but i cant copy or extract this higlighted text to another document...

¿there are one method or shortcut to do that? i try CTRL+SHIFT+L, but not work.

Thanks in advance, and congratulations for this good vs extension.

**Comment by RobHolme** [comment]
Hi,

The Highlight Field command only applies a background decoration, it does not select the field. I thought of making a change to also select the fields, but it wouldn't help you with dealing with multiple messages - the highlighting is only applied to a single file at a time as they become the active file in the editor. The highlighting is removed from files that are no longer the active tab. This might give the illusion that all files are highlighted, but is only a single file at a time (to avoid excessive memory consumption when a large amount of highlighting is applied).

I end up using using a powershell module (https://github.com/RobHolme/HL7-Powershell-Module) to extract specific fields when I need to deal with a large  volume of file hl7 files. The powershell module assumes some familiarity with powershell, and is command line only, so is not so user friendly as the VS code extension (it's also currently a Windows only solution).

Would it help if I provided a new function to extract/copy specified fields from all open messages? I'm short on free time this week, but if I find time over the weekend I might look into implementing something like this. 

Rob.

**Comment by RobHolme** [comment]
Hi,

I've finally found time to resolve the last issue I had in adding this feature - I've published an updated version of the extension (v.1.7.0)
 
The command 'HL7 Tools: Extract Fields from all Messages' will prompt you to enter a field location (e.g. PID-3), the corresponding values from all open files will be displayed in an output window. 

To work around a quirk with Visual Studio Code, the function needs to cycle through all open files the first time the command is run during a session. This may be slow if you have a lot of files open (the Powershell module I mentioned earlier will still be a better option if dealing with 100's of files). 

Feel free to give it a test - let me know if you have any issues.

Rob.
